test/test_ssl.py

```diff
@@ -225,6 +225,71 @@ def test_create_urllib3_context_ssl_version_and_ssl_min_max_version_no_error(
         ):
             ssl_.create_urllib3_context(**kwargs)
 
+    @pytest.mark.parametrize(
+        ["pha", "expected_pha", "cert_reqs"],
+        [
+            (None, None, None),
+            (None, None, ssl.CERT_NONE),
+            (None, None, ssl.CERT_OPTIONAL),
+            (None, None, ssl.CERT_REQUIRED),
+            (False, False, None),
+            (False, False, ssl.CERT_NONE),
+            (False, False, ssl.CERT_OPTIONAL),
+            (False, False, ssl.CERT_REQUIRED),
+            (True, False, None),
+            (True, False, ssl.CERT_NONE),
+            (True, False, ssl.CERT_OPTIONAL),
+            (True, False, ssl.CERT_REQUIRED),
+        ],
+    )
+    def test_create_urllib3_context_http2_no_pha(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        pha: bool | None,
+        expected_pha: bool | None,
+        cert_reqs: int | None,
+    ) -> None:
+        context = mock.create_autospec(ssl_.SSLContext)
+        context.set_ciphers = mock.Mock()
+        context.options = 0
+        context.post_handshake_auth = pha
+        monkeypatch.setattr(ssl_, "SSLContext", lambda *_, **__: context)
```

Comment:
If you are talking about mock for set_ciphers, I agree with you, but this test is mostly a copy of already existing `test_create_urllib3_context_pha`, but check for the opposite in case of http2 mode. I was not sure why it was there in the first place, so copied as is. Maybe we should remove it from both?

Reply:
I tried removing the mock for `test_create_urllib3_context_pha`, but it's actually needed as with a real `SSLContext` we can't set `post_handshake_auth` to None (it means `False`) or delete the attribute. Resolving, thanks, as a mock *is* needed.